### PR TITLE
chore: Enable `csharp_prefer_braces`

### PR DESCRIPTION
**Pull request by fabiomadge**
Previously we relied on enforcing this at review time. This PR adds a warning in the IDE and fixes all current violations. `dotnet-format` can only enforce this when analysing a whole project, which is surprisingly expensive (30s), so we don't make changes to the `pre-commit` hook at this time.
`dotnet tool run dotnet-format -s error -w Source/Dafny.sln`

### CODE REVIEW

**Comment by keyboardDrummer** [comment]
> we don't make the enabling changes at this time.

What do you mean by that? We want to run the full dotnet-format check in the build server, right?

**Comment by fabiomadge** [comment]
I'm not sure what happens in CI, but my comments is regarding the `pre-commit` hook.

**Comment by keyboardDrummer** [comment]
> I'm not sure what happens in CI, but my comment is regarding the `pre-commit` hook.

Could you turn on the full check in CI? For the pre-commit hook, I haven't used it but it might be worthwhile to check for opinions.

File: .github/workflows/msbuild.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       run: dotnet tool restore
     - name: Check code style
       working-directory: dafny
-      run: dotnet tool run dotnet-format -w --check Source/Dafny.sln --exclude Dafny/Scanner.cs --exclude Dafny/Parser.cs
+      run: dotnet tool run dotnet-format -s warn -w --check Source/Dafny.sln
```
Comment (keyboardDrummer) [inline .github/workflows/msbuild.yml:28]: Why did you remove the excludes?

Comment (fabiomadge) [reply .github/workflows/msbuild.yml:28]: I'm trying to somehow fix the errors that only come up in CI. Any suggestions are very welcome.

**Review by keyboardDrummer — APPROVED** [review]

File: Test/comp/CSharpStyling2.cs [hunk 2]
```diff
@@ -1,37 +1,49 @@
...
-      this.u = x;
-    }
-    public MyClass(bool y) {
-      this.u = y ? 3 : -3;
-    }
+namespace _module
+{
+    public partial class MyClass
+    {
+        public MyClass() { }  // needed by Dafny in order to call constructor "Init"
```
Comment (RustanLeino) [inline Test/comp/CSharpStyling2.cs:7]: All of these `Test/comp/*.cs` files have a different formatting. For example, the (new) code uses 4 spaces for indent and places many `{` on the next line for an `if`. Do the same code-style rules perhaps not apply in the same way for the `Test/comp` directory?

Comment (fabiomadge) [reply Test/comp/CSharpStyling2.cs]: Very good point. No, they didn't, only the files in `Source` were affected by `.editorconfig`. I moved it and reran `dotnet-format`. We don't check files beyond the scope of `Source/Dafny.sln` in the CI. Should we?

**Review by RustanLeino — APPROVED** [review]
